ldms/src/ldmsd/ldmsd_config.c

```diff
@@ -771,6 +771,214 @@ int __process_config_file(const char *path, int *lno, int trust,
+		if (!nline) {
+			rc = errno;
+			ldmsd_log(LDMSD_LERROR, "Out of memory\n");
+			goto cleanup;
+		}
+		line = nline;
+		line_sz = ((cnt + off)/line_sz + 1) * line_sz;
+	}
+	off += snprintf(&line[off], line_sz, "%s", tmp);
+
+	/* attempt to merge multiple lines together */
+	if (off > 0 && line[off-1] == '\\') {
+		line[off-1] = ' ';
+		goto next_line;
+	}
+
+parse:
+	if (!off)
+		goto next_line;
+
+	if (ldmsd_is_initialized()) {
+		if ((0 == strncmp(line, "prdcr_add", 9)) ||
+				(0 == strncmp(line, "prdcr_start", 11))) {
+			if (strstr(line, "interval")) {
+				ldmsd_log(LDMSD_LWARNING,
+						"'interval' is being deprecated. "
+						"Please use 'reconnect' with 'prdcr_add' or 'prdcr_start*' "
+						"in the future.\n");
+			}
+		}
+	}
+
+	req_array = ldmsd_parse_config_str(line, lineno, xprt.max_msg, ldmsd_log);
+	if (!req_array) {
+		rc = errno;
+		ldmsd_log(LDMSD_LERROR, "Process config string error in line\n %s\n "
+				". %s\n", line, STRERROR(rc));
+		goto cleanup;
+	}
+
+	request = __aggregate_records(req_array);
+	if (!request) {
+		rc = errno;
+		goto cleanup;
+	}
+	ldmsd_req_array_free(req_array);
+	req_array = NULL;
+
+	if (!ldmsd_is_initialized()) {
+		/* Process only the priority commands, e.g., cmd-line options */
+		if (!is_req_id_priority(ntohl(request->req_id)))
+			goto next_req;
+	} else {
+		/* Process non-priority commands, e.g., cfgobj config commands */
+		if (is_req_id_priority(ntohl(request->req_id)))
+			goto next_req;
+	}
+
+	/*
+	 * Make sure that LDMSD will create large enough buffer to receive
+	 * the config data.
+	 */
+	if (xprt.max_msg < ntohl(request->rec_len))
+		xprt.max_msg = ntohl(request->rec_len);
+
+	rc = ldmsd_process_config_request(&xprt, request, req_filter, ctxt);
+	if (rc || xprt.rsp_err) {
+		if (!rc)
+			rc = xprt.rsp_err;
+		ldmsd_log(LDMSD_LERROR, "Configuration error at line %d (%s)\n",
+				lineno, cfg_str);
+		goto cleanup;
+	}
+next_req:
+	free(request);
+	request = NULL;
+	off = 0;
+	goto next_line;
+
+cleanup:
+	if (buff)
+		free(buff);
+	if (line)
+		free(line);
+	if (lno)
+		*lno = lineno;
+	ldmsd_req_array_free(req_array);
+	if (request)
+		free(request);
+	return rc;
+}
+
+char *__process_yaml_config_file(const char *path, const char *dname)
+{
+	FILE *fp;
+	char command[256];
+	char cstr[256];
+        char *cfg_str = malloc(512);
+	*cfg_str = '\0';
+	snprintf(command, sizeof(command), "ldmsd_yaml_parser --ldms_config %s --daemon_name %s 2>&1", path, dname);
+	fp = popen(command, "r");
+	if (!fp)
+		printf("Error in yaml_parser\n");
+	size_t char_cnt = 0;
+	int status;
+	while (fgets(cstr, sizeof(cstr), fp) != NULL) {
+		char_cnt += strlen(cstr);
+		if (char_cnt > 512) {
+			cfg_str = (char *)realloc(cfg_str, char_cnt+1);
+			if (cfg_str == NULL) {
+				goto err;
+			}
+		}
+		strcat(cfg_str, cstr);
+	}
+	status = pclose(fp);
+	if (status)
+		goto err;
+	return cfg_str;
+err:
+	if (cfg_str) {
+		printf("ERROR: %s", cfg_str);
```

Comment:
We don't want all of cfg_str printed when there is an error. We just want to know what the error was from ldmsd_yaml_parser. When ldmsd_yaml_parser is fixed to write its errors to stderr, then we don't even need to worry about capturing that error message in ldmsd; it will already have gone to the right place.

Reply:
All good points, thanks for the review. This should also be taken care of in 33deed8. I'll implement fixes for some of the other issues you mentioned today.